oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/geometry/OSHDBGeometryBuilder.java

```diff
@@ -420,6 +420,8 @@ private static void cutAtSegment(LinkedList<OSMNode> ring, Segment cutSegment) {
         ring.add(ring.removeFirst());
       }
     }
+    assert false : "cut segment not found in ring";
+    throw new RuntimeException("cut segment not found in ring");
```

Comment:
Is this something you want to address? _Define and throw a dedicated exception instead of using a generic one._

Reply:
Right. I replace it with an `IllegalStateException`.